### PR TITLE
Add D9 compatibility to module

### PR DESCRIPTION
**Pull request by bia-wtag**
# Background
This module is not currently Drupal 9 compatible as shown in the `upgrade_status` report. For upgrading, the dependencies are updated, deprecated code was removed and info.yml was updated.

# Goal
- Module is D9 compatible.

# Acceptance
- Functionalities work.
- No error from `upgrade_status` and d9 compatible.

# Workflow
# Local testing
# Site Usable Link

### CODE REVIEW

File: src/LazyBulderCommentsStatistics.php [hunk 1]
```diff
@@ -26,7 +27,7 @@ class LazyBulderCommentsStatistics {
   /**
    * Constructs a new LazyBuilders object.
    */
-  public function __construct(EntityManagerInterface $entity_manager, CommentStatisticsInterface $comment_statistics) {
+  public function __construct(EntityTypeManagerInterface $entity_manager, CommentStatisticsInterface $comment_statistics) {
```
Comment (adi-wtag) [inline src/LazyBulderCommentsStatistics.php:30]: When are you creating object of this class ? 
I think It has link to https://github.com/wearewondrous/extra_fields_collection/blob/d9-compatibility/src/ExtraFieldsCollectionServiceProvider.php#L32
And this entity.manager should be entity_type.manager.

File: src/LazyBulderCommentsStatistics.php [hunk 2]
```diff
@@ -12,7 +13,7 @@ class LazyBulderCommentsStatistics {
   /**
    * Drupal\statistics\NodeStatisticsDatabaseStorage definition.
    *
-   * @var \Drupal\Core\Entity\EntityManagerInterface
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
    */
   protected $entityManager;
```
Comment (adi-wtag) [inline src/LazyBulderCommentsStatistics.php]: We can also change the name of the variable to `entityTypeManager`. But that's not mandatory.

**Review by adi-wtag — CHANGES_REQUESTED** [review]
Please, look for other entries of `entity.manager` when you make change.

File: src/LazyBulderCommentsStatistics.php (hunk 1 above)
Comment (bia-wtag) [reply src/LazyBulderCommentsStatistics.php:30]: This class is never instantiated. Here, in https://github.com/wearewondrous/extra_fields_collection/blob/d9-compatibility/src/ExtraFieldsCollectionServiceProvider.php#L32, the Reference class takes a string as a param which references a service. I have updated it.

File: src/LazyBulderCommentsStatistics.php (hunk 2 above)
Comment (bia-wtag) [reply src/LazyBulderCommentsStatistics.php]: Updated. Thanks for pointing out

File: src/Plugin/ExtraField/Display/ViewUrl.php [hunk 3]
```diff
@@ -107,8 +109,7 @@ public function getSetting($name) {
 
     if (is_callable('parent::getSetting')) {
       $value = $settings = parent::getSetting($name);
-    }
-    else {
+    } else {
```
Comment (zar-wtag) [inline src/Plugin/ExtraField/Display/ViewUrl.php]:
```suggestion
    } 
    else {
```
The previous way was as per drupal standard. Please keep it as it is.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+vendor
```
Comment (zar-wtag) [inline .gitignore]:
```suggestion
/vendor/
```
shouldnt it be the folder reference?

**Review by zar-wtag — APPROVED** [review]
I have placed couple of comments. Please review. Overall looks good.

File: src/Plugin/ExtraField/Display/ViewUrl.php (hunk 3 above)
Comment (bia-wtag) [reply src/Plugin/ExtraField/Display/ViewUrl.php]: This has happened due to my vscode formatter, It automatically formatted this when I saved. Reverting back in next commit.

File: .gitignore (hunk 4 above)
Comment (bia-wtag) [reply .gitignore]: yes bhai

**Review by bas-wtag — APPROVED** [review]

**Review by the-wtag — APPROVED** [review]

**Pull request closed without merging**